packages/contracts-bedrock/test/L2/SuperchainERC20.t.sol

```diff
@@ -0,0 +1,292 @@
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+import { Vm } from "forge-std/Vm.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { stdStorage, StdStorage } from "forge-std/Test.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+
+// Target contract
+import {
+    SuperchainERC20,
+    CallerNotBridge,
+    RelayMessageCallerNotL2ToL2CrossDomainMessenger,
+    MessageSenderNotThisSuperchainERC20,
+    CallerNotBridge
+} from "src/L2/SuperchainERC20.sol";
+import { ISuperchainERC20 } from "src/L2/ISuperchainERC20.sol";
+
+/// @title SuperchainERC20Test
+/// @dev Contract for testing the SuperchainERC20 contract.
+contract SuperchainERC20Test is Test {
+    string internal constant NAME = "SuperchainERC20";
+    string internal constant SYMBOL = "SCE";
+    uint8 internal constant DECIMALS = 18;
+    address internal constant BRIDGE = Predeploys.L2_STANDARD_BRIDGE;
+    address internal constant MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
+
+    SuperchainERC20 public superchainERC20;
+
+    /// @dev Sets up the test suite.
+    function setUp() public {
+        superchainERC20 = new SuperchainERC20(NAME, SYMBOL, DECIMALS);
+    }
+
+    /// @dev Helper function to setup a mock and expect a call to it.
+    function _mockAndExpect(address _receiver, bytes memory _calldata, bytes memory _returned) internal {
+        vm.mockCall(_receiver, _calldata, _returned);
+        vm.expectCall(_receiver, _calldata);
+    }
+
+    /// @dev Test that the bridge's constructor sets the correct values.
+    function test_constructor_succeeds() public view {
+        assertEq(superchainERC20.name(), NAME);
+        assertEq(superchainERC20.symbol(), SYMBOL);
+        assertEq(superchainERC20.decimals(), DECIMALS);
+    }
+
+    /// @dev Tests the `mint` function reverts when the caller is not the bridge.
+    function testFuzz_mint_callerNotBridge_reverts(address _caller, address _to, uint256 _amount) public {
+        // Ensure the caller is not the bridge
+        vm.assume(_caller != BRIDGE);
+
+        // Expect the revert with `CallerNotBridge` selector
+        vm.expectRevert(abi.encodeWithSelector(CallerNotBridge.selector));
+
+        // Call the `mint` function with the non-bridge caller
+        vm.prank(_caller);
+        superchainERC20.mint(_to, _amount);
+    }
+
+    /// @dev Tests the `mint` function reverts when the amount is zero.
+    function testFuzz_mint_zeroAddressTo_reverts(uint256 _amount) public {
+        // Expect the revert reason "ERC20: mint to the zero address"
+        vm.expectRevert("ERC20: mint to the zero address");
+
+        // Call the `mint` function with the zero address
+        vm.prank(BRIDGE);
+        address _to = address(0);
```

Comment:
check if there's a ZERO_ADDRESS constant, or create it otherwise and replace, you can probably do some:
``superchainERC20.mint({_to: ZERO_ADDRESS, _amount})` kind of thing to have it as a one liner